presto-spi/src/main/java/com/facebook/presto/spi/CatalogSchemaTableName.java

```diff
@@ -14,38 +14,25 @@
 
 package com.facebook.presto.spi;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonValue;
-
 import java.util.Objects;
 
 import static com.facebook.presto.spi.SchemaUtil.checkNotEmpty;
+import static java.util.Objects.requireNonNull;
 
-public class CatalogSchemaTableName
+public final class CatalogSchemaTableName
 {
     private final String catalogName;
     private final SchemaTableName schemaTableName;
 
-    @JsonCreator
-    public static CatalogSchemaTableName valueOf(String catalogSchemaTableName)
-    {
-        checkNotEmpty(catalogSchemaTableName, "catalogSchemaTableName");
-        String[] parts = catalogSchemaTableName.split("\\.");
-        if (parts.length != 3) {
-            throw new IllegalArgumentException("Invalid catalogSchemaTableName " + catalogSchemaTableName);
-        }
-        return new CatalogSchemaTableName(parts[0], parts[1], parts[2]);
-    }
-
     public CatalogSchemaTableName(String catalogName, SchemaTableName schemaTableName)
     {
-        this.catalogName = catalogName;
-        this.schemaTableName = schemaTableName;
+        this.catalogName = checkNotEmpty(catalogName, "catalogName");
+        this.schemaTableName = requireNonNull(schemaTableName, "schemaTableName is null");
     }
 
     public CatalogSchemaTableName(String catalogName, String schemaName, String tableName)
     {
-        this.catalogName = catalogName;
+        this.catalogName = checkNotEmpty(catalogName, "catalogName");
```

Comment:
Another minor issue is that it's generally best to have the constructor delegate and only have a single constructor that does field assignments. This makes it harder to miss a validation, since everything goes through the same path.